### ISSUE TITLE
Error trying to clean files at end of running the Diagnostics.bat on windows

### ISSUE THREAD

**Issue by pl853** [elastic/support-diagnostics#677]
Dear Elastic, 

First of all; thanks for the awesome tool, it's very usefull! 

In version 8.5.0 of the diagnostic tool, an error occurs at the end of the scanning process. Once the scanning is done, the script seems to clean up all the unnecessary files. However, when trying to clean up the diagnostics.log it returns an error, since the diagnostics.log is still open and being written to. 

![image](https://github.com/elastic/support-diagnostics/assets/24454699/68842e29-52f5-4ab9-a9ed-d2d0d13a6d6c)

I hope this can be fixed! Thanks in advance

Kind regards

**Comment by romain-chanu** [comment]
We have been able to reproduce this behaviour with the following set-up:
- Windows Server 2019
- OpenJDK 21
- Diagnostics utility version 8.5.0
- ESS deployment to generate a diagnostics in `api` mode

**TL;DR**
The error related to `diagnostics.log` does not prevent the generation of the diagnostics. However, it will prevent the utility to cleanup the temporary files.

**Investigation**

1) As per the report from @pl853 , below errors will be observed:
```
02:54:44.538 [main] INFO  co.elastic.support.BaseService - Closing loggers.
02:54:44.538 [main] INFO  co.elastic.support.BaseService - Archiving diagnostic results.
02:54:44.711 [main] INFO  co.elastic.support.util.ArchiveUtils - Archive: C:\Users\romain\Downloads\diagnostics-8.5.0-dist\diagnostics-8.5.0\api-diagnostics-20231208-025444.zip was created
02:54:44.836 [main] ERROR co.elastic.support.util.SystemUtils - Delete of directory:C:\Users\romain\Downloads\diagnostics-8.5.0-dist\diagnostics-8.5.0\api-diagnostics failed. Usually this indicates a permission issue
org.apache.commons.io.IOExceptionList: C:\Users\romain\Downloads\diagnostics-8.5.0-dist\diagnostics-8.5.0\api-diagnostics
	at org.apache.commons.io.FileUtils.cleanDirectory(FileUtils.java:331) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.commons.io.FileUtils.deleteDirectory(FileUtils.java:1192) ~[commons-io-2.11.0.jar:2.11.0]
	at co.elastic.support.util.SystemUtils.nukeDirectory(SystemUtils.java:51) [diagnostics-8.5.0.jar:8.5.0]
	at co.elastic.support.diagnostics.DiagnosticService.exec(DiagnosticService.java:98) [diagnostics-8.5.0.jar:8.5.0]
	at co.elastic.support.diagnostics.DiagnosticApp.main(DiagnosticApp.java:51) [diagnostics-8.5.0.jar:8.5.0]
Caused by: java.io.IOException: Cannot delete file: C:\Users\romain\Downloads\diagnostics-8.5.0-dist\diagnostics-8.5.0\api-diagnostics\diagnostics.log
	at org.apache.commons.io.FileUtils.forceDelete(FileUtils.java:1344) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.commons.io.FileUtils.cleanDirectory(FileUtils.java:324) ~[commons-io-2.11.0.jar:2.11.0]
	... 4 more
Caused by: java.nio.file.FileSystemException: C:\Users\romain\Downloads\diagnostics-8.5.0-dist\diagnostics-8.5.0\api-diagnostics\diagnostics.log: The process cannot access the file because it is being used by another process
	at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92) ~[?:?]
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103) ~[?:?]
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108) ~[?:?]
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:273) ~[?:?]
	at sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:109) ~[?:?]
	at java.nio.file.Files.deleteIfExists(Files.java:1191) ~[?:?]
	at org.apache.commons.io.file.PathUtils.deleteFile(PathUtils.java:487) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.commons.io.file.PathUtils.delete(PathUtils.java:392) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.commons.io.FileUtils.forceDelete(FileUtils.java:1341) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.commons.io.FileUtils.cleanDirectory(FileUtils.java:324) ~[commons-io-2.11.0.jar:2.11.0]
	... 4 more
```

2) We can confirm that the problem is not related to Windows Defender (given the above folder added in the exclusion list, the problem still occurs). 

3) At a glance in the code, these are some points that need to be further investigated (and fixed):
- `diagnostics.log` is does not tied-up to any [specific appender](https://github.com/elastic/support-diagnostics/blob/5b8cf1769d8cf1f6690667d8e606c61f90f54e4c/src/main/java/co/elastic/support/diagnostics/DiagnosticService.java#L90), 
- `diag` [appender](https://github.com/elastic/support-diagnostics/blob/5b8cf1769d8cf1f6690667d8e606c61f90f54e4c/src/main/resources/log4j2.xml#L22) does not seem to be use anywhere in the code
- given the above, `closeLogs` [method](https://github.com/elastic/support-diagnostics/blob/main/src/main/java/co/elastic/support/BaseService.java#L29) will not close the appender / `diagnostics.log` file 
- `nukeDirectory` [method](https://github.com/elastic/support-diagnostics/blob/main/src/main/java/co/elastic/support/util/SystemUtils.java#L47) triggers an exception given that the appender / `diagnostics.log` file is still opened

4) It has been observed  as well that using the default `Extract all...` in Windows reports that the generated zip file is invalid:

<img width="518" alt="Screenshot 2023-12-08 at 11 37 31 AM" src="https://github.com/elastic/support-diagnostics/assets/51113389/24107b2f-b434-4c9f-975c-8697a875ea1c">

```
Windows cannot complete the extraction.
The Compressed (zipped) Folder 'C:\api-diagnostics-20231208-023550.zip' is invalid.
```

It is unclear why this is happening. The zip file is definitely valid and can be opened using third-party tools like 7zip.

**Comment by geekpete** [comment]
Sometimes zip formats are slightly different and less compatible clients fail on them.
A previous issue with the windows extractor: https://github.com/elastic/support-diagnostics/issues/517#issuecomment-924463327
